gateway/src/main/java/io/pravega/example/iot/gateway/DataHandler.java

```diff
@@ -83,4 +93,41 @@ public String postData(@Context Request request, String data, @PathParam("remote
             throw e;
         }
     }
+
+    @POST
+    @Path("/rawData/{remoteAddr}")
+    public String postRawData(@Context Request request, byte[] data, @PathParam("remoteAddr") String remoteAddr) throws Exception {
+        final long ingestTimestamp = System.currentTimeMillis();
+        final String ingestTimestampStr = dateFormat.format(new Date(ingestTimestamp));
+        try {
+            if(data == null)
+                return "{DATA Not Received}";
+
+            // Get or calculate the routing key.
+            final String routingKeyAttributeName = Parameters.getRoutingKeyAttributeName();
```

Comment:
Simplify this section to use the routingKey value from the URL path. Do not use Parameters.getRoutingKeyAttributeName() and do not use Math.random().

Reply:
removed